samcli/commands/deploy/auth_utils.py

```diff
@@ -101,6 +101,13 @@ def _auth_id(resources_dict, event_properties, identifier):
     """
     resource_name = event_properties.get(identifier, "")
     api_resource = resources_dict.get(resource_name, {})
+
+    # Auth does not apply to ApiGateway::RestApi resources so return true and continue
+    if api_resource and (
```

Comment:
We have some constants here that you may be able to re-use https://github.com/aws/aws-sam-cli/blob/develop/samcli/lib/utils/resources.py